### DISCUSSION TITLE
useFieldArray show default field value, when it mounts after being unmounted, instead of last actual field value

### DISCUSSION THREAD

**Discussion by CyberNinja6610** [react-hook-form/react-hook-form#5204] [Q&A] [answered]
### Expected behavior
Field has actual value, when it mounts after being unmounted

### Expected behavior
Field value drops to default value, when it mounts after being unmounted

**Codesandbox link**
https://codesandbox.io/s/react-hook-form-usefieldarray-forked-904hv

Edit field and use "Show/Hide" button, to mount/unmount.

**Comment by bluebill1049** [comment]
<img width="1154" alt="Screen Shot 2021-05-15 at 10 41 59 pm" src="https://user-images.githubusercontent.com/10513364/118361425-c3ee5900-b5ce-11eb-8021-df180d42b027.png">

**Reply by CyberNinja6610** [reply]
Sorry, I've just corrected this error. You can look again.

**Reply by bluebill1049** [reply]
<img width="630" alt="Screen Shot 2021-05-15 at 10 54 42 pm" src="https://user-images.githubusercontent.com/10513364/118361866-8b4f7f00-b5d0-11eb-8ab2-4f962c281979.png">

please read the doc for correct syntax for `register` https://react-hook-form.com/api/usefieldarray

**Reply by CyberNinja6610** [reply]
I made it, but still have this problem

![изображение](https://user-images.githubusercontent.com/79763068/118362130-a1951500-b5a7-11eb-9274-95cab1207f1d.png)

**Comment by CyberNinja6610** [comment]
@bluebill1049, you can see this problem in this example from doc https://codesandbox.io/s/usefieldarray-conditional-2wi6f.
Just fill "name" with "bill", then input something in "conditional". Change "name" to something different to "bill", and then fill it with "bill" again. Value in "conditional" is lost.

**Comment by leapful** [comment, marked as answer]
@CyberNinja6610 

You are missing "defaultValue" for your "conditional" field.

Working sample: https://codesandbox.io/s/usefieldarray-conditional-forked-qvc6p?file=/src/App.tsx

**Reply by bluebill1049** [reply]
thank you @leapful
